### ISSUE TITLE
[BUG] Package contextvars_2.4-1 3003/3004 same version but different

### ISSUE THREAD

**Issue by timsen4711** [saltstack/salt#61203]
**Description**
Packages contextvars_2.4-1 in 3003/3004 have same version-string but they are different. Problem with package management (we are using aptly to manage our repos).

**Steps to Reproduce the behavior**

![image](https://user-images.githubusercontent.com/67151651/140909704-1b73fce6-5b1b-4a30-bfee-64a1f56a4949.png)

aptly@aptly01-a> stat /srv/aptly/repo/pool/8c/c8/a2383376f5390b09be2dbbd8904d_python3-contextvars_2.4-1_all.deb
  File: /srv/aptly/repo/pool/8c/c8/a2383376f5390b09be2dbbd8904d_python3-contextvars_2.4-1_all.deb
  Size: 4536            Blocks: 16         IO Block: 4096   regular file
Device: fd02h/64770d    Inode: 2099146     Links: 2
Access: (0644/-rw-r--r--)  Uid: ( 1005/   aptly)   Gid: ( 1456/   aptly)
Access: 2021-11-08 15:37:17.078696246 +0100
Modify: 2021-10-12 22:15:27.402278636 +0200
Change: 2021-10-27 13:53:36.113218073 +0200
 Birth: -

aptly@aptly01-a> stat /srv/aptly/repo/pool/3c/dc/f650efa930c4ddc2ad79da11e48d_python3-contextvars_2.4-1_all.deb
  File: /srv/aptly/repo/pool/3c/dc/f650efa930c4ddc2ad79da11e48d_python3-contextvars_2.4-1_all.deb
  Size: 4532            Blocks: 16         IO Block: 4096   regular file
Device: fd02h/64770d    Inode: 4590507     Links: 1
Access: (0644/-rw-r--r--)  Uid: ( 1005/   aptly)   Gid: ( 1456/   aptly)
Access: 2021-11-09 10:49:07.574885457 +0100
Modify: 2021-11-09 10:49:07.574885457 +0100
Change: 2021-11-09 10:49:08.690890413 +0100
 Birth: -

**Comment by OrangeDog** [comment]
Presumably the 3004 version should be `2.4-2`?

**Comment by timsen4711** [comment]
This is right to stay compliant with the debian package convention and, as in this case, not to break the deduplication of a package management tool.

**Comment by bryceml** [comment]
those are for different distro versions, if we want to have them be unique among all packages that exist, we're going to need to start throwing the the distro and version in version somehow, possibly the architecture too, as there may be differences in packaging needed between those.

**Comment by OrangeDog** [comment]
Ah, I didn't spot that.

@timsen4711 there are Debian 10 repos for both 3003 and 3004.
They should have identical packages when the version is the same.

**Comment by bryceml** [comment]
With debian's repos, they just don't package new versions of stuff for oldoldstable, or oldstable, other than security fixes, so they don't run into this problem much.

I did see a lot of debian packages are starting to have `+deb10u2` like markings on them.  I also see that postgresql sticks something like `20.04` or `110` (debian 11) in the package version in order to be unique since they package many versions for many distros and distro versions.

I don't think this is worth changing now that we've been doing it this way for so long.  How much end user automation would we possibly mess up by changing this, for how much benefit.  Aptly users just need to keep things separate by distro/distro-version/arch.

The repos should have this principle already within a given distro/distro-version/arch folder on repo.saltproject.io.

**Comment by timsen4711** [comment]
Distro is not part to have a debian compliant package pool. It's only arch, name and version. If these three things match, the size of a package should also be the same regardless of the source it comes from. 
You just have to decide whether or not.
